tools/src/main/java/org/apache/kafka/tools/KafkaClientPerformance.java

```diff
@@ -52,72 +48,47 @@ public static void main(String[] args) throws Exception {
-            String payloadFilePath = res.getString("payloadFile");
-            String transactionalId = res.getString("transactionalId");
             boolean shouldPrintMetrics = res.getBoolean("printMetrics");
             long transactionDurationMs = res.getLong("transactionDurationMs");
+            long flushDurationMs = res.getLong("flushDurationMs");
+            boolean flushEnabled = 0 < flushDurationMs;
             boolean transactionsEnabled =  0 < transactionDurationMs;
 
-            // since default value gets printed with the help text, we are escaping \n there and replacing it with correct value here.
-            String payloadDelimiter = res.getString("payloadDelimiter").equals("\\n") ? "\n" : res.getString("payloadDelimiter");
-
             if (producerProps == null && producerConfig == null) {
                 throw new ArgumentParserException("Either --producer-props or --producer.config must be specified.", parser);
             }
 
-            List<byte[]> payloadByteList = new ArrayList<>();
-            if (payloadFilePath != null) {
-                Path path = Paths.get(payloadFilePath);
-                System.out.println("Reading payloads from: " + path.toAbsolutePath());
-                if (Files.notExists(path) || Files.size(path) == 0)  {
-                    throw new  IllegalArgumentException("File does not exist or empty file provided.");
-                }
-
-                String[] payloadList = new String(Files.readAllBytes(path), "UTF-8").split(payloadDelimiter);
-
-                System.out.println("Number of messages read: " + payloadList.length);
-
-                for (String payload : payloadList) {
-                    payloadByteList.add(payload.getBytes(StandardCharsets.UTF_8));
-                }
-            }
-
-            Properties props = new Properties();
-            if (producerConfig != null) {
-                props.putAll(Utils.loadProps(producerConfig));
-            }
-            if (producerProps != null)
-                for (String prop : producerProps) {
-                    String[] pieces = prop.split("=");
-                    if (pieces.length != 2)
-                        throw new IllegalArgumentException("Invalid property: " + prop);
-                    props.put(pieces[0], pieces[1]);
-                }
-
+            Properties props = generateProps(producerProps, producerConfig);
             props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArraySerializer");
             props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArraySerializer");
+
+            final String applicationId = "Customized-EOS-Performance-" + UUID.randomUUID();
             if (transactionsEnabled)
-                props.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, transactionalId);
+                props.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, applicationId);
```

Comment:
This makes the transactionalId non-deterministic. Is there a good reason to do that? I think we should always blow away the destination topic, so if we ever end up using the same transactionalId that would indicate a problem in our test setup?

Reply:
I see, the challenge here is that whether it's worth blowing away the input topic every time, as now the producer will send offsets for the progress tracking, which means for next test job to be run we need to clean up the old offset.

Reply:
How about we reset the offsets between tests? I think that may be what you're suggesting. It would be cleaner not to accumulate metadata over time.

Reply:
Sg

Reply:
While I was doing the work, I realized that our intent to delete/reset offset could fail. I still updated code to do offset reset as it's more metadata size bounded, just bring this up and see you feel about the two ways here.

Reply:
I've been doing the delete / reset offset path in all of my tests and it seems to be going fine. For the output topic I delete and do create in a loop because the list command sometimes does not show the topic but then create fails because it thinks it still exists. We could get the committed offset at the beginning and fail if it is greater than 0.

Reply:
Sg, I finally used a `seek` call to reset offset by the end.